### ISSUE TITLE
and yet again Travis OSX failures

### ISSUE THREAD

**Issue by KrisThielemans** [UCL/STIR#623]
we're getting the following type of error all the time since today

https://travis-ci.org/github/UCL/STIR/jobs/706598064#L8112-L8115
```
    from _stir import *
E   ImportError: dlopen(/Users/travis/build/UCL/STIR/install/python/_stir.so, 2): Library not loaded: /usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/Python
E     Referenced from: /Users/travis/build/UCL/STIR/install/python/_stir.so
E     Reason: image not found
```
this is somewhat similar to what we had https://github.com/SyneRBI/SIRF/issues/753, so I've put in a `brew upgrade` in https://github.com/UCL/STIR/pull/621. that gives 
https://travis-ci.org/github/UCL/STIR/jobs/706677710#L5307-L5311
```
Warning: python 3.8.3_2 already installed
ls: /usr/local/Cellar/python/3.*/Frameworks/Python.framework/Versions/3.*: No such file or directory
usage: basename string [suffix]
       basename [-a] [-s suffix] string [...]
Something wrong with finding Python executable for OSX
```
so it appears that `brew` installs Python now somewhere else and the following line no longer works.
https://github.com/UCL/STIR/blob/3503d735225ad52eef1f1c5f60a9211d44c66fdd/.travis.yml#L111

terrible

**Comment by rijobro** [comment]
I just had a look at my machine and it appears I have two versions with `brew`:

`/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/bin/python3.7`

and

`/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/bin/python3.8`

My guess is that they're not replacing 3.7 yet so they're putting the more recent 3.8 elsewhere. I guess that then means that if you don't have 3.7, that folder will be empty.

Why can we not simply use `PY_EXE=$(which python3)`? Even if you accidentally picked up the system version of `python3` (which we could avoid by prepending `/usr/local/bin` to `PATH`), it would still hopefully work.

**Comment by rijobro** [comment]
or `PY_EXE=/usr/local/bin/python${PYMVER}` would work with almost no risk as far as I can see.

**Comment by KrisThielemans** [comment]
not bad. what about `PY_EXE=$(which python${PYMVER})`?

**Comment by rijobro** [comment]
yup. only risk is that if `python@3.8` is keg only, it may not go into `/usr/local/bin`. But we can give it a go. If it works with this weird set up (3.8 is in a different location from normal), then it should work for all brew set ups.

**Comment by KrisThielemans** [comment]
I checked that the log says
```
Python has been installed as
  /usr/local/bin/python3
Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /usr/local/opt/python@3.8/libexec/bin
```
So I guess it'll be fine. To be honest, I don't care which python we pick up on Travis, as long as it works...

**Comment by rijobro** [comment]
Sure, I think any version is fine. We probably don't even need to brew install it, as there'll probably be a system `python3` anyway these days (in `/usr/bin/python3`).
